src/handwritten/_time.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+from tzlocal import get_localzone
+from datetime import timezone
+from datetime import tzinfo as dt_tzinfo
+from datetime import datetime as std_datetime
+from hightime import datetime as ht_datetime
+from typing import Optional, Union
+from zoneinfo import ZoneInfo
+
+# theoretically the same as astimezone(), but with support for dates before 1970
+def _convert_to_desired_timezone(expected_time_utc: Union[std_datetime, ht_datetime], tzinfo: Optional[dt_tzinfo] = None):
+    # if timezone matches, no need to do conversion
+    if expected_time_utc.tzinfo is tzinfo:
+        return expected_time_utc
+
+    # if timezone is not defined, use system timezone
+    if tzinfo is None:
+        tzinfo = get_localzone()
+
+    # use ZoneInfo here to account for daylight savings
+    if isinstance(tzinfo, ZoneInfo):
+        localized_time = expected_time_utc.replace(tzinfo=tzinfo)
+        desired_expected_time = tzinfo.fromutc(localized_time)
+        return(desired_expected_time)
```

Comment:
FYI, check for understanding: the original time is in UTC. `replace(tzinfo=)` changes the timezone without adjusting the time. `tzinfo.fromutc()` adjusts the time. So this emulates `astimezone` without hitting the 1970 case.

Reply:
yep you got it right :)